### ISSUE TITLE
Failure to install if a directory name contains a space(s)

### ISSUE THREAD

**Issue by masasakano** [ffi/ffi#623]
It seems installing via `gem install` or `bundle` fails if a part of the absolute path contains a space(s).  Here is an example (when installing with `bundle install` in `rails generate`).  In the log, the line `make: *** ["/It/is/My] Error 2` should be `["/It/is/My Drive/my_project/..."]` but the characters after the space are apparently ignored.

    Installing ffi 1.9.23 with native extensions
    Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
    
        current directory: /It/is/My Drive/my_project/vendor/bundle/ruby/2.5.0/gems/ffi-1.9.23/ext/ffi_c
    /Users/myname/.rbenv/versions/2.5.0/bin/ruby -r ./siteconf20180320-64149-lvpz0r.rb extconf.rb
    checking for ffi.h... no
    checking for ffi.h in /usr/local/include,/usr/include/ffi... yes
    checking for ffi_call() in -lffi... yes
    checking for ffi_closure_alloc()... no
    checking for ffi_raw_call()... no
    checking for shlwapi.h... no
    checking for rb_thread_blocking_region()... no
    checking for rb_thread_call_with_gvl()... yes
    checking for rb_thread_call_without_gvl()... yes
    creating extconf.h
    creating Makefile
    
    current directory: /It/is/My Drive/my_project/vendor/bundle/ruby/2.5.0/gems/ffi-1.9.23/ext/ffi_c
    make "DESTDIR=" clean
    
    current directory: /It/is/My Drive/my_project/vendor/bundle/ruby/2.5.0/gems/ffi-1.9.23/ext/ffi_c
    make "DESTDIR="
    /bin/sh: -c: line 0: unexpected EOF while looking for matching `"'
    /bin/sh: -c: line 1: syntax error: unexpected end of file
    make: *** ["/It/is/My] Error 2
    
    make failed, exit code 2
    
    Gem files will remain installed in /It/is/My Drive/my_project/vendor/bundle/ruby/2.5.0/gems/ffi-1.9.23 for inspection.
    Results logged to /It/is/My Drive/my_project/vendor/bundle/ruby/2.5.0/extensions/x86_64-darwin-17/2.5.0-static/ffi-1.9.23/gem_make.out
    
    An error occurred while installing ffi (1.9.23), and Bundler cannot continue.
    Make sure that `gem install ffi -v '1.9.23'` succeeds before bundling.
    
    In Gemfile:
      selenium-webdriver was resolved to 3.11.0, which depends on
        childprocess was resolved to 0.9.0, which depends on
          ffi
    $

**Comment by davidstosik** [comment]
I have the same issue. `ffi` is a dependency on my project, and when I run `bundle install --path={some path with spaces}`, then it fails with the errors you quoted. Running `bundle install --path=/tmp/bundle` works.

**Comment by chrisvasselli** [comment]
I just started having this issue as well, with ffi 1.15.1. I never had it before, so I'm not totally sure what caused it, but I recently deleted and reinstalled the Xcode command line tools, that may have been the trigger.

Updating my path to one without spaces seemed to resolve the issue, but I have a feeling that's not the bottom of the problem.

**Comment by chrisvasselli** [comment]
I spent some time trying to debug this, and came up short. But I did get a little closer.

It's failing during the make command, and I ran that command with the `-d` argument to get additional debugging information on both a machine where it's building successfully in a folder with a space, and a folder where it's not.

The difference happens while it's trying to generate the first file compiled file, AbstractMemory.o. On the machine that works, it considers a bunch of target files, and eventually prints

```
Finished prerequisites of target file `AbstractMemory.o'.
    Must remake target `AbstractMemory.o'.
```
and then goes on to compiling it.

But on the machine that doesn't work, it has another target it considers, which is the partial path, up to where the space is. It prints `Considering target file `"/Users/wedge/jenkins/workspace/Setup'` and then breaks down. (my path is /Users/wedge/jenkins/workspace/Setup Build Machine/)

I'm not an expert on makefiles, and I haven't been able to figure out why it's considering that path as a target on one machine, and not on the other. But I figured I would document my findings for any future people who try to diagnose the issue.

**Comment by chrisvasselli** [comment]
tl;dr this fixed the issue for me:
```
brew uninstall --ignore-dependencies pkg-config
brew install pkg-config
```

OK, so I went down a deep rabbit hole.

It looks like what's going on is that libffi cannot be built in a folder with a space in it. And as far as I can tell, that's known and accepted. But why do we need to build libffi in the first place? Because it's not properly finding the system libffi during the `ruby extconf.rb` call.

So, why is it not finding it? Because pkg_config is pointing it to the directory of libffi in the old Xcode developer tools:
```
% pkg-config --cflags libffi
-I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi
```

I just deleted that old version of the command line tools and installed the latest, which is why it can't find the headers there. So why is it looking there? Because pkg-config is looking in the wrong pkg-config directory for hombrew installations. We can see this by running `pkg-config --variable pc_path pkg-config`. On my machine, the output looked like:

`/opt/homebrew/lib/pkgconfig:/opt/homebrew/share/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/11`

That "11" refers to macOS 11, perhaps the version of macOS I had installed when I first installed pkg-config? If I look in that folder, I see that the libffi.pc in that folder lists cflags as `-I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi`.

So, I tried to figure out where that `/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/11` path was coming from, but using `fs_usage` I can see that pkg-config is not reading it from anywhere on disk.

So, could it have somehow been written into the environment, or binary, or something at install-time? That was my best-guess. So I uninstalled and reinstalled pkg-config from homebrew using the command at the top of this comment. And sure enough, the `pc_path` variable was updated to: `/opt/homebrew/lib/pkgconfig:/opt/homebrew/share/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib/pkgconfig:/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/13`.

And the above error has gone away.

It feels like there's some sort of bug here, where pkg-config should know to point to the latest folder, but I'm not sure if that's a bug in pkg-config, homebrew, the Xcode developer tools installer, or just somehow expected behavior.
